### PR TITLE
SSCS-2669: Implement Crossbrowser testing with Saucelabs

### PR DESCRIPTION
**Pull request by Rob-Doherty**
## WHAT?

This PR adds crossbrowser testing via saucelabs, that runs different browsers in parallel, and triggers in the overnight pipeline.

## WHY?

This automated test will exercise our service on a variety of browsers, reducing the amount of manual testing that needs to take place and improving confidence with automated releases.

## HOW?

- Jenkinsfile_nightly triggers tests in overnight pipeline, which calls out to `test:crossbrowser`
- can also be run locally using `yarn test:crossbrowser` with relevant SAUCE_USERNAME and SAUCE_ACCESS_KEY
- initially limited to run against Chrome and Firefox only
- crossbrowser tests will run in parallel browsers for speed
- includes SauceLabsReportingHelper.js for reporting test outcomes to saucelabs
- runs via WebdriverIO, using a separate CodecetJS config file: saucelabs.con.js

**JIRA:** SSCS-2669

### CODE REVIEW

File: test/e2e/saucelabs.conf.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+const setupConfig = {
+  tests: './smoke/appeal.test.js',
+  output: process.env.E2E_OUTPUT_DIR || config.get('e2e.outputDir'),
+  features: {
+    evidenceUpload: {
+      enabled: evidenceUploadEnabled
+    }
+  },
+  helpers: {
+    WebDriverIO: {
```
Comment (iguannago) [inline test/e2e/saucelabs.conf.js]: cannot we use Puppeteer?

File: test/e2e/saucelabs.conf.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        options: {
+          reportDir: process.env.E2E_OUTPUT_DIR || config.get('e2e.outputDir'),
+          reportName: 'index',
+          inlineAssets: true
+        }
+      }
+    }
+  },
+  multiple: {
+    // microsoft: {
```
Comment (iguannago) [inline test/e2e/saucelabs.conf.js]: if it's not used it's good to have a cleaner code and remove those commented out lines :)

File: Jenkinsfile_nightly [hunk 3]
```diff
@@ -0,0 +1,25 @@
+#!groovy
+
+properties([
+        pipelineTriggers([cron('00 20 * * *')]),
+        parameters([
+                string(name: 'URL_TO_TEST', defaultValue: 'https://benefit-appeal.nonprod.platform.hmcts.net', description: 'The URL you want to run these tests against'),
```
Comment (iguannago) [inline Jenkinsfile_nightly]: This is cool. I am wondering if we could use this same approach to run our smoke and functional tests in AAT against the external url

File: test/e2e/saucelabs.conf.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+      });
+    } else {
+      console.error('ERROR: supportedBrowsers.js is empty or incorrectly defined');
+    }
+  }
+  return browserConfig;
+};
+
+const setupConfig = {
+  tests: './smoke/appeal.test.js',
```
Comment (iguannago) [inline test/e2e/saucelabs.conf.js]: I guess we run this test just for now to check it all works fine and once that confirms we may want to add a more complete batch. Is that correct?

File: test/e2e/saucelabs.conf.js (hunk 1 above)
Comment (Rob-Doherty) [reply test/e2e/saucelabs.conf.js]: No unfortunately not, Saucelabs needs a remote webdriver instance to tell it what commands to execute, whereas puppeteer is a tool for Chrome/Chromium only

File: test/e2e/saucelabs.conf.js (hunk 4 above)
Comment (Rob-Doherty) [reply test/e2e/saucelabs.conf.js]: Thats a good point, I had been thinking of restricting the crossbrowser tests to just a single test that covers all / most of the pages, but I'm not familiar enough with SSCS yet to say which test that might be

Comment (Rob-Doherty) [reply test/e2e/saucelabs.conf.js]: If you have any ideas of a more suitable test / couple of tests then more than happy to change to use those instead

File: test/e2e/saucelabs.conf.js (hunk 2 above)
Comment (Rob-Doherty) [reply test/e2e/saucelabs.conf.js]: Thats a fair point, I was hoping they wouldn't be commented long term but you're right about cleaner code, I'll remove 👍

Comment (Rob-Doherty) [reply test/e2e/saucelabs.conf.js]: done

File: test/e2e/saucelabs.conf.js (hunk 4 above)
Comment (iguannago) [reply test/e2e/saucelabs.conf.js]: let's leave it as it is just for now, yes. :)

**Pull request merged**